grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/
+package org.pih.warehouse.invoice
+
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Party
+import org.pih.warehouse.core.UnitOfMeasure
+import org.pih.warehouse.core.User
+
+class Invoice implements Serializable {
+
+    def beforeInsert = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            createdBy = currentUser
+            updatedBy = currentUser
+        }
+    }
+
+    def beforeUpdate = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            updatedBy = currentUser
+        }
+    }
+
+    String id
+    String invoiceNumber
+    String referenceNumber
```

Comment:
Can you remind me what this is for?

Reply:
@jmiranda Should I remove it and leave only the invoiceNumber?

Reply:
No just add a hasMany for referenceNumber like shipment.
```
static hasMany = [referenceNumbers: ReferenceNumber]
```